### ISSUE TITLE
Behind a Strong Firewall and Proxy NPM Install Errors on Certain Libraries

### ISSUE THREAD

**Issue by raechelann-zz** [npm/npm#19381]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ X] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
I work for a big (very big) bank. I have been tasked with learning Angular for an upgrade of one of our legacy applications. I have completed most of the simple examples to learn basic angular concepts. However, when i run a simple npm install, npm is trying to pull the node-sass library(and others) from a server my proxy will not allow. I want to know how i can change the source location that npm installs its libraries from when i type npm install. If i can configure the location to point somewhere locally, I can pull the libraries internally and have it install from an internal shared server.

#### How can the CLI team reproduce the problem?

I just type (1) ng new Hello-World or (2)npm install in a newly created project (like Hello-World) and the error presents itself about 1/2 way thru. It is specific to the node-sass libraries. Once this file fails, several of the remaining steps fail. I am including the relevant error information below.

### supporting information:

**Error as seen in terminal window text:**
Downloading binary from https://github.com/sass/node-sass/releases/download/v4.7.2/win32-x64-59_binding.node
Cannot download "https://github.com/sass/node-sass/releases/download/v4.7.2/win32-x64-59_binding.node":

HTTP error 403 Forbidden

Hint: If github.com is not accessible in your location
      try setting a proxy via HTTP_PROXY, e.g.

      export HTTP_PROXY=http://example.com:1234

or configure npm proxy via

      npm config set proxy http://example.com:8080

> uglifyjs-webpack-plugin@0.4.6 postinstall C:\Work\Problem-Ticket\node_modules\webpack\node_modules\uglifyjs-webpack-plugin
> node lib/post_install.js

> node-sass@4.7.2 postinstall C:\Work\Problem-Ticket\node_modules\node-sass
> node scripts/build.js

**Relevant version information**

Relevant version info:
Angular CLI: 1.6.0
Node: 9.2.0
OS: win32 x64

Command I am running (output with error is attached):
ng new Hello-World

**** My Proxy information has been set using config commands but i cannot share that information because it is considered confidential. 

The bottom line: I need to be able to change where npm installs this from so I can put a local copy on our team server and have our developers install it from there. There has to be a way to configure WHERE this file is pulled from.

Thanks for any help you can give in advance.

Cheers,
Raechel

**Comment by kenany** [comment]
Note that this is something that `node-sass` is doing, not `npm` itself. Checkout the `scripts/install.js` script in the `node-sass` package. It uses the [`getBinaryUrl()`](https://github.com/sass/node-sass/blob/v4.7.2/lib/extensions.js#L237-L245) function to determine where to download binaries from. Looks like you can either use an environment variable, or set a custom config in `npm`'s own settings.

Since this is unrelated to `npm` itself, I'll close this, but hopefully the above information is enough to get you started.

**Comment by raechelann-zz** [comment]
Thank you. Yes, it gives me enough information to continue to work towards a resolution for us.
